bin/clang/util.cc

```diff
@@ -367,5 +367,16 @@ std::string replace_all(std::string const &str, std::string const& match,
   return result_string;
 }
 
+std::string replace_all(std::string const &str,
+    std::map<std::string, std::string> const& replacements) {
```

Comment:
@jjwilke do we want to try to template this on map like types? I guess we could template it on container and just get bad error messages if the type in the container doesn't contain `first` and `second` members that can be converted to `std::string const&`

Reply:
Yeah - that would be good. Do the LLVM types fit the concept?